### PR TITLE
Feature/gapdc 7 springboot security

### PR DESCRIPTION
**Pull request by praveenmurugan101**
- Replace default Spring Boot login form with a customized React form
- Configure Spring to allow public access to specific endpoints
- Create csrf endpoint to generate tokens for React form
- Configure Vite.js to clean and build files inside the static folder within the resources directory
- Set React form to post login details in the format expected by Spring Boot

[https://docs.spring.io/spring-security/reference/servlet/authentication/passwords/form.html](https://github.com/APHA-CSU/GenAPP/pull/url)

### CODE REVIEW

**Review by ellisrichardj — APPROVED** [review]
Great - looks good to me 👍

**Review by sainab-omar — APPROVED** [review]
Looks good 👍

File: src/main/java/gov/apha/genapp/restcontroller/AuthStatus.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package gov.apha.genapp.restcontroller;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
```
Comment (televisedbrain) [inline src/main/java/gov/apha/genapp/restcontroller/AuthStatus.java:10]: please add javadoc

File: src/main/java/gov/apha/genapp/restcontroller/AuthStatus.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package gov.apha.genapp.restcontroller;
```
Comment (televisedbrain) [inline src/main/java/gov/apha/genapp/restcontroller/AuthStatus.java:1]: could you please add the copyright headers to the top of new classes? thanks

File: src/main/java/gov/apha/genapp/restcontroller/Csrf.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package gov.apha.genapp.restcontroller;
+
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import jakarta.servlet.http.HttpServletRequest;
+
+import org.springframework.security.web.csrf.CsrfToken;
+
```
Comment (televisedbrain) [inline src/main/java/gov/apha/genapp/restcontroller/Csrf.java:9]: please add javadoc

File: src/main/java/gov/apha/genapp/restcontroller/Csrf.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+@RestController
+public class Csrf {
+    /**
+     * Endpoint to get csrf token which will be used in react login form
```
Comment (televisedbrain) [inline src/main/java/gov/apha/genapp/restcontroller/Csrf.java:13]:
```suggestion
     * Endpoint to get csrf token which will be used in react login form
     * @param request HTTP request
     * @return { add a description here }
```

**Review by televisedbrain — APPROVED** [review]
nice

**Pull request merged**